File: onlinejudge/baekjoon/1302.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Map<String, Integer> result = new HashMap<>();
        Scanner scan = new Scanner(System.in);
        int N = scan.nextInt();

        for (int i = 0; i < N; i++) {
            String input = scan.next();
            if (result.containsKey(input)) {
                result.put(input, result.get(input) + 1);
            } else {
                result.put(input, 1);
            }
        }

        String answer = null;
        int max = 0;
        for (var entry : result.entrySet()) {
            if (entry.getValue() > max) {
                answer = entry.getKey();
                max = entry.getValue();
            } else if (entry.getValue() == max) {
                List<String> list = new ArrayList<>();
                list.add(answer);
                list.add(entry.getKey());
                Collections.sort(list);
                answer = list.get(0);
            }
        }
        System.out.println(answer);
    }
}

File: onlinejudge/baekjoon/1302.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

func main() {
	result := make(map[string]int)
	scanner := bufio.NewScanner(os.Stdin)
	
	var N int
	fmt.Scan(&N)

	for i := 0; i < N; i++ {
		scanner.Scan()
		input := scanner.Text()
		result[input]++
	}

	var answer string
	max := 0
	for key, value := range result {
		if value > max {
			answer = key
			max = value
		} else if value == max {
			if key < answer {
				answer = key
			}
		}
	}
	fmt.Println(answer)
}

